Src/Microsoft.Scripting/PlatformAdaptationLayer.cs

```diff
@@ -41,6 +41,12 @@ public class PlatformAdaptationLayer {
         [Obsolete("This will be removed in the the future.")]
         public static readonly bool IsCompactFramework = false;
 
+        public static bool IsNativeModule { get; } = _IsNativeModule();
+
+        private static bool _IsNativeModule() {
+            return typeof(void).Assembly.GetType().FullName != "System.Reflection.RuntimeAssembly";
```

Comment:
Is there another way to check this? This give a false positive on Mono where `FullName` is `System.Reflection.MonoAssembly`...

Reply:
I'm not sure, but it can be replaced by
`typeof(void).Assembly.Modules.ElementAt(0).ToString() == "<Unknown>"`
https://github.com/dotnet/corert/blob/07fb8bdcae9cabb8f699bc89216dc58909c90e6f/src/System.Private.Reflection.Core/src/System/Reflection/Runtime/Modules/RuntimeModule.cs#L72
There is no more elegant approach, it is necessary to determine on indirect grounds.